contracts/src/RatesManager.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+contract RatesManager is AccessControlDefaultAdminRules {
+    // Create a dedicated role for managing rates
+    bytes32 public constant RATES_ADMIN_ROLE = keccak256("RATES_ADMIN_ROLE");
+
+    // Rates struct holds the fees and the start time of the rates
+    struct Rates {
+        uint64 messageFee;
+        uint64 storageFee;
+        uint64 congestionFee;
+        uint64 startTime;
+    }
+
+    // All Rates appended here
+    Rates[] private allRates;
```

Comment:
This gives us ordering, but on the other hand is pretty expensive when we're looking for an specific timestamp in the past.

What about a `mapping(uint64 => Rates) private ratesByTimestamp;` and controlling wether to add the next timestamp with a `_lastTimestampAdded` or a cronological array `uint64[] private timestamps;`?
That would enable us to have a `getRateAt(timestamp)`

Additionally, the array is unbounded, we could consider pruning older timestamps when adding new ones, so the array doesn't grow beyond a certain point.

Reply:
My plan is to ingest all of them into the node and put them in a nice in-memory data structure there.

We do need all historical rates so that new nodes can get the correct Rates when processing very old messages.